flink-core/src/main/java/org/apache/flink/api/java/typeutils/PojoTypeInfo.java

```diff
@@ -317,7 +308,39 @@ public int getFieldIndex(String fieldName) {
 
 		return new PojoSerializer<T>(getTypeClass(), fieldSerializers, reflectiveFields, config);
 	}
-	
+
+	@Override
+	@PublicEvolving
+	public <F> FieldAccessor<T, F> getFieldAccessor(String fieldExpression, ExecutionConfig config) {
+		
+		FieldAccessor.FieldExpressionDecomposition decomp = FieldAccessor.decomposeFieldExpression(fieldExpression);
```

Comment:
What about renaming `FieldExpressionDecomposition` to just `FieldExpression`? IMHO the old class name is too long.